pkgs/tools/package-management/nix/default.nix

```diff
@@ -60,17 +60,26 @@ let
     sha256 = "sha256-s1ybRFCjQaSGj7LKu0Z5g7UiHqdJGeD+iPoQL0vaiS0=";
   };
 
+  versions =
+    lib.foldl lib.attrsets.unionOfDisjoint {}
+      (lib.pipe
+        (builtins.readDir ./.) [
+          (lib.mapAttrsToList (k: _: k))
+          (lib.lists.filter (k: builtins.match "[0-9]+\\.[0-9]+\\.nix" k != null))
+          (map (f: callPackage (import (./. + "/${f}")) { inherit common patch-sqlite-exception boehmgc-nix_2_3 patch-monitorfdhup; }))
+        ]);
```

Comment:
Also I am not convinced on this change also because it creates more complexity which I think we don't really need or want.